client/src/main/java/org/apache/celeborn/client/write/DataPusher.java

```diff
@@ -130,8 +125,10 @@ public void run() {
                 }
               } catch (CelebornIOException e) {
                 exceptionRef.set(e);
-              } catch (InterruptedException | IOException e) {
+              } catch (IOException e) {
                 exceptionRef.set(new CelebornIOException(e));
+              } catch (InterruptedException e) {
+                logger.error("DataPusher push thread interrupted while pushing data.");
```

Comment:
Seems if interrupted, the while loop will never finish because ```stillRunning``` will not return true.

Reply:
Should break here

Reply:
> Should break here

Done